File: player/src/tiny/scene/layer/layer.cpp
#include <tiny/scene/layer/layer.h>
#include <tiny/scene/node/node.h>

namespace Tiny {

Layer::Layer(Scene *_scene, const uint8_t *_definition, Node **_nodes, uint8_t _numberOfNodes) :
    scene(_scene), definition(_definition), nodes(_nodes), numberOfNodes(_numberOfNodes) {
}

void Layer::draw() {
  for (uint8_t i = 0; i < numberOfNodes; i++) {
    Node *node = nodes[i];
    node->draw();
  }
}

}

File: export/lib/model/project/entity/scene/layer.js
const Entity = require('../../entity');
const array = require('../array');

const filtered = (layer, type) => array(layer.project.sortBy(layer.models.filter(model => model.baseType === type), 'json.index', 'asc'));

class Layer extends Entity {

  get baseType() {
    return 'scene/layer';
  }

  bind() {
    super.bind();
    this.index = this.parent.layers.indexOf(this);
    this.nodes = filtered(this, 'scene/layer/node');
  }

}

module.exports = Layer;
